autode/values.py

```diff
@@ -65,23 +68,21 @@ def _to(value: Union["Value", "ValueArray"], units: Union[Unit, str]):
             f"No viable unit conversion from {value.units} -> {units}"
         )
 
-    # Convert to the base unit, then to the new units
-    c = float(units.conversion / value.units.conversion)
-
-    if isinstance(value, Value):
-        return value.__class__(float(value) * c, units=units)
-
-    elif isinstance(value, ValueArray):
-        value[:] = np.array(value, copy=True) * c
-        value.units = units
-        return value
-
-    else:
+    if not (isinstance(value, Value) or isinstance(value, ValueArray)):
         raise ValueError(
             f"Cannot convert {value} to new units. Must be one of"
             f" Value of ValueArray"
         )
 
+    # Convert to the base unit, then to the new units
+    c = float(units.conversion / value.units.conversion)
+
+    new_value = value if inplace else value.copy()
+    new_value *= c
+    new_value.units = units
+
+    return None if inplace else new_value
```

Comment:
@t-young31 Would it be better to make it so that exception is raise if inplace=True is called for `value` as inplace doesn't make sense? I realise that `_to()` is a module private function so it should really not be called from outside, but still for the sake of understanding of a programmer who looks at the code or perhaps as a guard against incorrect call. Because right now calling value._to(unit, inplace=True) will just return None without doing anything.

Reply:
yes, it would. will throw in an `assert` for clarity